api/account/api.go

```diff
@@ -85,13 +120,54 @@ func (user AccountRepresentation) Validate() error {
 	return nil
 }
 
+// Validate is a validator for UpdatePasswordBody
+func (updatePwd UpdatePasswordBody) Validate() error {
+	if !matchesRegExp(updatePwd.CurrentPassword, RegExpPassword) {
+		return errors.New("Invalid current Password")
+	}
+
+	if !matchesRegExp(updatePwd.NewPassword, RegExpPassword) {
+		return errors.New("Invalid new Password")
+	}
+
+	if !matchesRegExp(updatePwd.ConfirmPassword, RegExpPassword) {
+		return errors.New("Invalid confirm Password")
+	}
+
+	return nil
+}
+
+// Validate is a validator for CredentialRepresentation
+func (credential CredentialRepresentation) Validate() error {
+	if credential.Id != nil && !matchesRegExp(*credential.Id, RegExpID) {
+		return errors.New("Invalid Id")
+	}
+
+	if credential.Type != nil && !matchesRegExp(*credential.Type, RegExpType) {
+		return errors.New("Invalid Type")
+	}
+
+	if credential.UserLabel != nil && !matchesRegExp(*credential.UserLabel, RegExpLabel) {
+		return errors.New("Invalid Label")
+	}
+
+	return nil
+}
+
 func matchesRegExp(value, re string) bool {
 	res, _ := regexp.MatchString(re, value)
 	return res
 }
 
 // Regular expressions for parameters validation
 const (
+	RegExpID         = `^[a-z0-9]{8}-[a-z0-9]{4}-[a-z0-9]{4}-[a-z0-9]{4}-[a-z0-9]{12}$`
+	RegExpIDNullable = `^([a-z0-9]{8}-[a-z0-9]{4}-[a-z0-9]{4}-[a-z0-9]{4}-[a-z0-9]{12})|(null)$`
```

Comment:
We want to support the explicit value (null) for null values ?

Reply:
we use it only for credential move. If we move a credential at the top of the list, we call moveAfter with null as there is nothing before.